### PR TITLE
Add asyncOpenURL wrapper

### PR DESCRIPTION
**Pull request by luacmartins**
cc @chiragsalian @roryabraham 

### Fixed Issues
$ https://github.com/Expensify/Expensify.cash/issues/3615

### Tests
1. Send a `Request Money` request from Account A to Account B (both accounts must have PayPal.me username set in `Settings > Payments`)
2. Open Safari and log in to Account B.
3. From Account B, click Pay and from the Dropdown select Pay with `PayPal.me` and then confirm the payment. 
4. A new window should open and 
4. Verify that you are redirected to Paypal.me and the handle is that of the user you owe and amount matches.

### QA Steps
Steps above.

### Tested On

- [X] Web
- [X] Mobile Web
- [X] Desktop
- [X] iOS
- [X] Android

#### Web

https://user-images.githubusercontent.com/22219519/123491035-f7ab9e00-d5d2-11eb-9b5c-fd553da32b4d.mov

#### Mobile Web

https://user-images.githubusercontent.com/22219519/123680754-4bea9400-d806-11eb-8a0d-aa479002a0f0.mov

#### Desktop

https://user-images.githubusercontent.com/22219519/123680775-5147de80-d806-11eb-8c77-697ac3baf754.mov

#### iOS

https://user-images.githubusercontent.com/22219519/123680958-776d7e80-d806-11eb-9f3b-ed1a992bd014.mov

#### Android

https://user-images.githubusercontent.com/22219519/123680814-5c027380-d806-11eb-9d39-4714c7e869cc.mov

### CODE REVIEW

**Comment by roryabraham** [comment]
Slack thread for context: https://expensify.slack.com/archives/C01GTK53T8Q/p1623955103415500

cc @rushatgabhane

File: src/libs/asyncOpenURL/index.website.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import {Linking} from 'react-native';
+
+/**
+ * Prevents Safari from blocking pop-up window when opened within async call.
+ *
+ * @param {Promise} promise
+ */
+export default function asyncOpenURL(promise) {
+    const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
```
Comment (roryabraham) [inline src/libs/asyncOpenURL/index.website.js:9]: Can you explain why are you doing a negative patten match for chrome or android here? Is there some case when the user agent would be formatted like 'chrome-xxx-safari' or something?

File: src/libs/asyncOpenURL/index.desktop.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import {Linking} from 'react-native';
```
Comment (roryabraham) [inline src/libs/asyncOpenURL/index.desktop.js:1]: It's been a while so I'm not 100% sure, but does it work to provide an index.website.js for the web implementation and just index.js for all other platforms?

File: src/libs/asyncOpenURL/index.website.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import {Linking} from 'react-native';
+
+/**
+ * Prevents Safari from blocking pop-up window when opened within async call.
+ *
+ * @param {Promise} promise
+ */
+export default function asyncOpenURL(promise) {
```
Comment (roryabraham) [inline src/libs/asyncOpenURL/index.website.js:8]: Also, what do you think about just providing the url as an argument to this function? Is that simpler/less prescriptive than this implementation, in which the promise needs to resolve to a url? Basically something more like [this implementation](https://expensify.slack.com/archives/C01GTK53T8Q/p1624062070455500?thread_ts=1623955103.415500&channel=C01GTK53T8Q&message_ts=1624062070.455500)

**Comment by roryabraham** [comment]
Also, we should make sure to test on all platforms to ensure we don't introduce a regression on other platforms

File: src/libs/asyncOpenURL/index.website.js (hunk 1 above)
Comment (rushatgabhane) [reply src/libs/asyncOpenURL/index.website.js:9]: @roryabraham I think all third party browsers on iOS are just skins for safari.
So chrome on iOS has this issue too. And this pattern detects that in my testing :) 

The negative pattern match is to exclude both chrome and edge, and android.

Edge and Chrome's userAgent: `"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36"`

Comment (luacmartins) [reply src/libs/asyncOpenURL/index.website.js:9]: Thanks for answering this @rushatgabhane! Yes, that is the reason I used the negative pattern match. @roryabraham Do we have any other preferred method?

File: src/libs/asyncOpenURL/index.desktop.js (hunk 2 above)
Comment (luacmartins) [reply src/libs/asyncOpenURL/index.desktop.js:1]: @roryabraham It seems to work! Would that be preferred? If so, I will also go ahead and update the docs [here](https://github.com/Expensify/Expensify.cash#platform-specific-file-extensions)

File: src/libs/asyncOpenURL/index.website.js (hunk 3 above)
Comment (luacmartins) [reply src/libs/asyncOpenURL/index.website.js]: Updated!

**Comment by luacmartins** [comment]
Updated with requested changes and tests on all platforms.

**Review by roryabraham — APPROVED** [review]
Okay, looks better to me.

**Review by chiragsalian — APPROVED** [review]

**Review by Luke9389 — APPROVED** [review]

**Comment by roryabraham** [comment]
Unfortunately the deploy comments are not working right now. This was deployed to staging yesterday.

**Pull request merged**